pandas/core/groupby/generic.py

```diff
@@ -770,6 +772,7 @@ def value_counts(
         mask = ids != -1
         ids, val = ids[mask], val[mask]
 
+        lab: Index | np.ndarray
```

Comment:
This pattern happens quite often because of mypy's early binding assumption (first declaration, even if only in one branch, determines the type). I prefer pyright's behavior:

```py
if foo():
    a = 1 # mypy: a is of type int (in this branch, in all other branches, and after the if/else)
    # pyright: a is of type int in this branch
else: 
    a = "1" # mypy error!
    # pyright: a is of type str in this branch
# pyright: a is of type int | str
```